### PR TITLE
docs: multiregion deployment feature

### PR DESCRIPTION
**Pull request by tgross**
To go with https://github.com/hashicorp/nomad/pull/8184

### CODE REVIEW

**Comment by angrycub** [comment]
saw an &emdash; leaking out to the output.    Not sure how to fix yet or would have made a suggestion.  Just wanted to let you know I was looking.

**Comment by tgross** [comment]
Fixed the emdash in 89eca7f

**Comment by tgross** [comment]
@yishan-lin @changli0617 it'd probably be worth getting some of your input here with respect to the nomenclature gap between "Multi-Cluster Deployment" and "multiregion deployment" as it's been crafted in the implementation. The operator-facing concept in Nomad is the `region` (see also [`region`](https://www.nomadproject.io/docs/job-specification/job/#region) and the [Multi-Region Federation Guide](https://learn.hashicorp.com/nomad/operating-nomad/federation)), which is why we used the configuration value `multiregion`. 

It's probably fine that we have the "marketing name" in the enterprise section of the docs here so long as it points to the operator-facing name, but I wanted to bring it to your attention.

File: website/pages/docs/enterprise/index.mdx [hunk 1]
```diff
@@ -86,6 +86,18 @@ When a Nomad cluster is at capacity for a given set of placement constraints, an
...
 
+## Multi-Cluster & Efficiency
+
+Multi-Cluster & Efficiency features are part of an add-on module that enables
+an organization to operate Nomad at scale across multiple clusters through
+features such as Multiregion Deployments.
+
+### Multiregion Deployments
+
+Multiregion Deployments enable an operator to deploy a single job to multiple
```
Comment (schmichael) [inline website/pages/docs/enterprise/index.mdx:97]: Link to stanza

Comment (tgross) [reply website/pages/docs/enterprise/index.mdx]: Done in 28299df

**Comment by tgross** [comment]
Previews:
* https://deploy-preview-8185--nomad-website.netlify.app/docs/job-specification/multiregion/
* https://deploy-preview-8185--nomad-website.netlify.app/docs/enterprise/#multi-cluster-efficiency
* https://deploy-preview-8185--nomad-website.netlify.app/docs/commands/deployment/unblock/

**Pull request merged**